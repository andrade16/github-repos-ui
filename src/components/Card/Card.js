import React from "react";
import Avatar from "../Avatar/Avatar";
import "./Card.scss";

const Card = ({
  title,
  link,
  subtitle,
  icon,
  buttonLabel,
  buttonToolTip,
  avatarUrl,
  dataObj,
  subIcon,
  onClick,
}) => {
  const onCommitClick = () => {
    console.log(dataObj);
    onClick(dataObj);
  };

  return (
    <>
      <div className="card">
        <Avatar
          src={avatarUrl}
          width={150}
          height={150}
          icon={icon}
        />
        <div className="card-subicon">
          {subIcon} {subtitle}
        </div>
        <div className="card-body">
          <h4 className="card-title">{title}</h4>
          <p className="card-content">
            <a className="card-link" href={link}>
              {link}
            </a>
          </p>
          <button
            title={buttonToolTip}
            className="card-action"
            onClick={onCommitClick}
          >
            {buttonLabel}
          </button>
        </div>
      </div>
    </>
  );
};

export default Card;
