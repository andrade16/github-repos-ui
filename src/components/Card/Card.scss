.card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 0.25rem;

  .card-icon,
  .card-subicon {
    display: flex;
    justify-content: center;
  }

  .card-subicon {
    padding: 10px;
    align-items: center;
  }

  .card-body {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    flex: 1 1;
    padding: 1.25rem;
    background-color: aliceblue;
    word-wrap: break-word;

    .card-title {
      margin-bottom: 0.75rem;
    }

    .card-content {
      margin-top: 0;
      margin-bottom: 1rem;

      .card-link {
        color: #007bff;
        text-decoration: none;
        background-color: transparent;
      }
    }

    .card-action {
      cursor: pointer;
      color: #fff;
      background-color: #007bff;
      border-color: #007bff;
      border-radius: 0.25rem;
      height: 5vh;
      width: 10vw;
      min-width: 128px;
    }
  }
}
