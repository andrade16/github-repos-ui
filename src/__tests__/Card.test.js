import React from "react";
import renderer from "react-test-renderer";
import { FaGithub } from "react-icons/fa";
import { AiFillStar } from "react-icons/ai";
import Card from "../components/Card/Card";


const mockDataObj = {
  full_name: "test obj",
  html_url: "https://google.com",
  stargazers_count: 400,
  owner: {
    avatar_url: "https://avatars.githubusercontent.com/u/16903644?v=4",
  },
};

describe("<Card/>", () => {
  it("Renders <Card/> correctly", () => {
    const tree = renderer
      .create(
        <Card
          title={mockDataObj.full_name}
          link={mockDataObj.html_url}
          subtitle={`Stars: ${mockDataObj.stargazers_count}`}
          buttonLabel={"View commits"}
          buttonToolTip={"View commits made in the last 24 hours"}
          avatarUrl={mockDataObj.owner.avatar_url}
          icon={<FaGithub style={{ width: 150, height: 150, marginTop: 5 }} />}
          subIcon={<AiFillStar style={{ color: "yellow" }} />}
          dataObj={mockDataObj}
          onClick={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
