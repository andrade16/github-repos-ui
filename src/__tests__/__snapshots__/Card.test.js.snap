// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> Renders <Card/> correctly 1`] = `
<div
  className="card"
>
  <div
    className="avatar"
  >
    <img
      alt="avatar-image"
      className="avatar-image"
      src="https://avatars.githubusercontent.com/u/16903644?v=4"
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
  </div>
  <div
    className="card-subicon"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "yellow",
        }
      }
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
      />
    </svg>
     
    Stars: 400
  </div>
  <div
    className="card-body"
  >
    <h4
      className="card-title"
    >
      test obj
    </h4>
    <p
      className="card-content"
    >
      <a
        className="card-link"
        href="https://google.com"
      >
        https://google.com
      </a>
    </p>
    <button
      className="card-action"
      onClick={[Function]}
      title="View commits made in the last 24 hours"
    >
      View commits
    </button>
  </div>
</div>
`;
